fix(lsp): handle optional requests to avoid MethodNotFound (-32601) with MATLAB Language Server

csacca opened the pull request:
# Fix: Handle optional LSP requests to avoid `MethodNotFound` (-32601) crashing LSP server (`matlab-language-server`)

Fixes #4003

## Summary
- Add no-op handlers for `client/registerCapability` and `client/unregisterCapability`.
- Implement `workspace/workspaceFolders` response aligned with the initialize root.
- Eliminates `MethodNotFound` (-32601) noise and improves interoperability with `matlab-language-server`.

## Motivation
Servers may issue optional LSP requests; unhandled methods cause `MethodNotFound` errors and crash `matlab-language-server`. Handling these requests is low-risk and LSP spec-aligned.

## Changes
- packages/opencode/src/lsp/client.ts
  - Register handlers before `connection.listen()`:
    - `connection.onRequest("client/registerCapability", async () => ({} as any))`
    - `connection.onRequest("client/unregisterCapability", async () => ({} as any))`
    - `connection.onRequest("workspace/workspaceFolders", async () => [{ name: "workspace", uri: "file://" + input.root }])`

## Tests
- packages/opencode/test/lsp/client.test.ts
  - Covers all three requests using `packages/opencode/test/fixture/lsp/fake-lsp-server.js`.

## Manual validation
- Launch `matlab-language-server` via `opencode.json` (see #4003 for configuration details); open a `.m` file; verify no `MethodNotFound` (-32601) for the above methods; diagnostics flow; LSP server does not crash.

csacca committed:
fix(lsp): handle optional requests to avoid MethodNotFound (-32601) with MATLAB Language Server

- Add no-op handlers for client/registerCapability and client/unregisterCapability
- Implement workspace/workspaceFolders response aligned with initialize root
- Add tests using a fake LSP server to cover all three requests

Fixes #4003

rekram1-node commented:
i think this one should be good to go can u fix the ci failure

rekram1-node commented on packages/opencode/src/lsp/client.ts:
```diff
@@ -62,6 +62,14 @@ export namespace LSPClient {
       // Return server initialization options
       return [input.server.initialization ?? {}]
     })
+    connection.onRequest("client/registerCapability", async () => ({} as any))
```
can we replace these with:
```
    connection.onRequest("client/registerCapability", async () => {})
```

csacca replied:
Ok, removed the `any` casts and ran `scripts/format.ts` on all three files - should pass CI now.

csacca committed:
chore(lsp): remove unnecessary any-casts and apply formatting

- Replace no-op handlers to return undefined per maintainer feedback
- Apply repository formatting (client and tests)
